tools/extra/eosio_abigen_tool/eosio-abigen.cpp.in

```diff
@@ -340,6 +350,25 @@ class abigen : public generation_utils {
             if (as.name == t.type)
                return true;
          }
+         for( auto td : _abi.typedefs ) {
+            if (as.name == remove_suffix(td.type))
+               return true;
+         }
+         return false;
+      };
+      
+      auto validate_types = [&]( abi_typedef td ) {
+         for ( auto as : _abi.structs )
+            if (validate_struct(as))
+               for ( auto f : as.fields )
+                  if ( f.type == td.new_type_name )
+                     return true;
+         for ( auto t : _abi.tables )
+            if ( t.type == td.new_type_name )
+               return true;
+         for ( auto a : _abi.actions )
+            if ( a.type == td.new_type_name )
+               return true;
```

Comment:
I think we might need to add
```      
      for ( auto t : _abi.typedefs ) {
            if (td.new_type_name == remove_suffix(t.type))
               return true;
         }
```
This will fixes the nested typedef case such as:
```
      struct nested_alias { int v; };
      typedef nested_alias nested_alias_1;
      typedef nested_alias_1 nested_alias_2;
```